torch/autograd/variable.py

```diff
@@ -358,6 +359,21 @@ def __dir__(self):
         keys = variable_methods + attrs
         return sorted(keys)
 
+    # Numpy array interface, to support `numpy.asarray(tensor) -> ndarray`
+    def __array__(self, dtype=None):
+        if dtype is None:
+            return self.cpu().numpy()
```

Comment:
If the variable is on the GPU, is the implicit CPU conversion here to be expected by the user?

Reply:
It's copied as-is from Tensor.py, but I also think this is a reasonable choice.